.github/workflows/publish-docs.yml

```diff
@@ -35,6 +35,13 @@ jobs:
       - name: Generate API Docs
         run: python code_doc_autogen.py
 
```

Comment:
Maybe we can explicitly call pre-commit before the verification step but after code_doc_autogen so that we get a well-formatted snapshot that's consistent with what's checked into master.

```
- name: Install Pre-commit
   run: pip install pre-commit
- name: Run pre-commit
   run: pre-commit run --all-files
```

Reply:
Yes, this sounds like a great idea @justinxzhao . That way the mark down files that will be used to generate the website match what someone saw on their local!